lib/service/computetask.go

```diff
@@ -1038,6 +1062,48 @@ func (s *ComputeTaskService) getInputAsset(kind asset.AssetKind, key, identifier
 	return nil, orcerrors.NewUnimplemented(fmt.Sprintf("unsupported input kind: %q", kind.String()))
 }
 
+// getTaskWorker will determine the worker on which the task should execute
+func (s *ComputeTaskService) getTaskWorker(input *asset.NewComputeTask, algo *asset.Algo) (string, error) {
+	var (
+		inferredWorker string
+	)
+	for _, input := range input.Inputs {
+		algoInput, ok := algo.Inputs[input.Identifier]
+		if !ok {
+			return "", orcerrors.NewInvalidAsset(fmt.Sprintf("unknown task input: this identifier was not declared in the Algo: %q", input.Identifier))
+		}
+		if algoInput.Kind != asset.AssetKind_ASSET_DATA_MANAGER {
+			continue
+		}
+
+		dm, err := s.getCachedDataManager(input.GetAssetKey())
+		if err != nil {
+			return "", err
+		}
+
+		if inferredWorker != "" && inferredWorker != dm.Owner {
+			return "", orcerrors.NewInvalidAsset("Task cannot depend on two datamanager with different owners")
+		}
+
+		inferredWorker = dm.Owner
```

Comment:
We don't support multiple data managers

```suggestion
		return dm.Owner, nil
```